packages/picker/test/index.ts

```diff
@@ -2362,4 +2365,180 @@ export function runPickerTests(): void {
             ).to.be.greaterThan(-1);
         });
     });
+    describe('icons attribute', () => {
+        it('hides icon in button when icons="none"', async () => {
+            const el = await fixture<Picker>(html`
+                <sp-picker label="Choose an action" value="1" icons="none">
+                    <sp-menu-item value="1">
+                        <sp-icon-edit slot="icon"></sp-icon-edit>
+                        Edit
+                    </sp-menu-item>
+                    <sp-menu-item value="2">
+                        <sp-icon-copy slot="icon"></sp-icon-copy>
+                        Copy
+                    </sp-menu-item>
+                </sp-picker>
+            `);
+            await elementUpdated(el);
+            await nextFrame();
+            await nextFrame();
+
+            const iconSpan = el.shadowRoot.querySelector(
+                '#icon'
+            ) as HTMLElement;
+            expect(iconSpan).to.not.be.null;
+            expect(iconSpan.hidden, 'icon span should be hidden').to.be.true;
+
+            // Verify the label is still visible
+            const labelSpan = el.shadowRoot.querySelector(
+                '.label'
+            ) as HTMLElement;
+            expect(labelSpan).to.not.be.null;
+            expect(
```

Comment:
This assertion is not required since you are already checking labelSpan for non-null